packages/contracts/contracts/engine/V3/EngineFactoryV0.sol

```diff
@@ -0,0 +1,215 @@
+            IEngineFactoryV0.EngineCoreType.Engine
+            ? engineImplementation
+            : engineFlexImplementation;
+
+        engineContract = Clones.clone({implementation: implementation});
+
+        IGenArt721CoreContractV3_Engine(engineContract).initialize(
+            engineConfiguration,
+            adminACLContract
+        );
+
+        // update super admin address if not null address
+        if (engineConfiguration.newSuperAdminAddress != address(0)) {
+            address[] memory genArt721CoreAddressesToUpdate = new address[](1);
+            genArt721CoreAddressesToUpdate[0] = engineContract;
+
+            IAdminACLV0(adminACLContract).changeSuperAdmin(
+                engineConfiguration.newSuperAdminAddress,
+                genArt721CoreAddressesToUpdate
+            );
+        }
+
+        string memory coreType = IGenArt721CoreContractV3_Engine(engineContract)
+            .coreType();
+        string memory coreVersion = IGenArt721CoreContractV3_Engine(
+            engineContract
+        ).coreVersion();
+        // register the new Engine contract
+        ICoreRegistryV1(coreRegistry).registerContract(
+            engineContract,
+            stringToBytes32(coreVersion),
+            stringToBytes32(coreType)
+        );
+        // emit event
+        emit EngineContractCreated(engineContract);
+    }
+
+    /**
+     * @notice Abandons the contract, preventing it from being used to create
+     * new Engine and Engine Flex contracts.
+     * Only callable by the owner, and only once; reverts otherwise.
+     */
+    function abandon() external onlyOwner {
+        require(!isAbandoned, "factory is abandoned");
+        // set isAbandoned to true
+        isAbandoned = true;
+        // emit event
+        emit Abandoned();
+    }
+
+    /**
+     * @notice Drains the contract's balance to the `recipient`.
+     * @param recipient The address to send funds to.
+     * Only callable by the owner.
+     */
+    function drainETH(address payable recipient) external onlyOwner {
+        uint256 balance = address(this).balance;
+        if (balance > 0) {
+            (bool success, ) = recipient.call{value: balance}("");
+            require(success, "Payment failed");
+        }
+    }
+
+    /**
+     * @notice Drains the contract's balance of an input ERC20 token to
+     * the `recipient`.
+     * @param ERC20TokenAddress The address of the ERC20 token to withdraw.
+     * @param recipient The address to send ERC20 tokens to.
+     * Only callable by the owner.
+     */
+    function drainERC20(
+        address ERC20TokenAddress,
+        address recipient
+    ) external onlyOwner {
+        IERC20 token = IERC20(ERC20TokenAddress);
+        uint256 balance = token.balanceOf(address(this));
+        if (balance > 0) {
+            SafeERC20.safeTransfer({
+                token: token,
+                to: recipient,
+                value: balance
+            });
+        }
+    }
+
```

Comment:
I think it makes sense to implement something that allows us to arbitrarily execute calls from this contract, in case third parties make weird assumptions about contract "deployers" being eligible for airdrops, it would prevent loss of funds. Should be doable following something like this: https://github.com/0xSplits/splits-contracts-monorepo/blob/dfaf6f5cfa02ca96989b00ed183f053e44964ae8/packages/splits-v2/src/utils/Wallet.sol#L54
(note: don't recommend using that example directly, as we don't want to broadcast that this contract is _intended_ to actually hold NFTs)

Reply:
Nice! I ported the function over but removed `value` from the struct as I don't think claims typically need any value sent in the tx 🤔  Do you think it's worth any clarifying comments that this is intended to prevent loss of funds?

Reply:
Awesome 🙏 Updated the comment here